common/php/class-module.php

```diff
@@ -5,296 +5,338 @@
-	 */
-	function get_all_post_types() {
+			return isset( $edit_flow->$slug ) && $edit_flow->$slug->is_enabled();
+		}
 
-		$allowed_post_types = array(
-			'post' => __( 'Post' ),
-			'page' => __( 'Page' ),
-		);
-		$custom_post_types = $this->get_supported_post_types_for_module();
+		/**
+		 * Returns whether vip features have been enabled or not.
+		 *
+		 * @since 0.10.0
+		 *
+		 * @return true, if the module is enabled, false otherwise
+		 */
+		protected function are_vip_features_enabled() {
+			global $edit_flow;
 
-		foreach( $custom_post_types as $custom_post_type => $args ) {
-			$allowed_post_types[$custom_post_type] = $args->label;
+			return 'on' === $edit_flow->settings->module->options->vip_features;
 		}
-		return $allowed_post_types;
-	}
 
-	/**
-	 * Cleans up the 'on' and 'off' for post types on a given module (so we don't get warnings all over)
-	 * For every post type that doesn't explicitly have the 'on' value, turn it 'off'
-	 * If add_post_type_support() has been used anywhere (legacy support), inherit the state
-	 *
-	 * @param array $module_post_types Current state of post type options for the module
-	 * @param string $post_type_support What the feature is called for post_type_support (e.g. 'ef_calendar')
-	 * @return array $normalized_post_type_options The setting for each post type, normalized based on rules
-	 *
-	 * @since 0.7
-	 */
-	function clean_post_type_options( $module_post_types = array(), $post_type_support = null ) {
-		$normalized_post_type_options = array();
-		$all_post_types = array_keys( $this->get_all_post_types() );
-		foreach( $all_post_types as $post_type ) {
-			if ( ( isset( $module_post_types[$post_type] ) && $module_post_types[$post_type] == 'on' ) || post_type_supports( $post_type, $post_type_support ) )
-				$normalized_post_type_options[$post_type] = 'on';
-			else
-				$normalized_post_type_options[$post_type] = 'off';
+		/**
+		 * Returns whether vip features have been enabled or not.
+		 *
+		 * @since 0.10.0
+		 *
+		 * @return true, if the module is enabled, false otherwise
+		 */
+		protected function is_analytics_enabled() {
+			global $edit_flow;
+
+			return 'on' === $edit_flow->settings->module->options->analytics;
 		}
-		return $normalized_post_type_options;
-	}
 
-	/**
-	 * Get all of the possible post types that can be used with a given module
-	 *
-	 * @param object $module The full module
-	 * @return array $post_types An array of post type objects
-	 *
-	 * @since 0.7.2
-	 */
-	function get_supported_post_types_for_module( $module = null ) {
-
-		$pt_args = array(
+		/**
+		 * Check if the site is a WPVIP site.
+		 *
+		 * @since 0.10.0
+		 * @return true, if it is a WPVIP site, false otherwise
+		 */
+		protected function is_vip_site() {
+			return defined( 'WPCOM_IS_VIP_ENV' ) && constant( 'WPCOM_IS_VIP_ENV' ) === true
+			&& defined( 'WPCOM_SANDBOXED' ) && constant( 'WPCOM_SANDBOXED' ) === false
+			&& defined( 'FILES_CLIENT_SITE_ID' );
```

Comment:
@ingeniumed I noticed in `vip dev-env` that `WPCOM_IS_VIP_ENV` is `false`, it seems like it's only set to `true` on production sites. Can we change this to use the presence of `VIP_GO_ENV` instead? The valid values for this seem to be `"production"` and `"local"`, but I think just checking that it exists will do the trick:

```suggestion
			return defined( 'VIP_GO_ENV' )
			&& defined( 'WPCOM_SANDBOXED' ) && constant( 'WPCOM_SANDBOXED' ) === false
			&& defined( 'FILES_CLIENT_SITE_ID' );
```

Note that this is different than [the logic we used in the block data API](https://github.com/Automattic/vip-block-data-api/blob/05134af5f95a9fe055a205374943a2a842f35698/src/analytics/analytics.php#L99-L104), and that's mostly because we didn't want to send analytics locally. However, I think we should have the "Turn on WordPress VIP features" default to true on `vip dev-env`.

That probably means we'll want two VIP-related checks, one for analytics (exclude local environments), and one for VIP features (local environments are fine).

Reply:
Good catch, corrected this in [this commit](https://github.com/Automattic/Edit-Flow/pull/732/commits/60f04b0c947fcdcf95423049e0e7278963abf892) by allowing for a flag to be passed in that will ensure only production sites are considered wpvip sites.